### PR TITLE
Start the trial budget at frame zero, not at `reset`

### PR DESCRIPTION
**Pull request by vertix**
## Summary

A trial's deadline was armed when `task.reset()` returned, but a simulated producer's `reset` only *arms* frame zero — the proxy publishes it on its next turn, and that turn advances the virtual clock by one control period without stepping the env. So every trial spent a control period of its budget before its episode had started, and an env's own N-step horizon landed at `(N+1)·dt` on the harness clock.

The harness already knows this. `TelemetryScope.start_rollout` anchors a rollout's virtual duration at the first control cycle that has an observation, for exactly this reason, and two tests pin that behaviour. The deadline was the last consumer still measuring from `reset`.

- Re-anchor `_deadline` to the episode's first complete observation, where an env that paces the trial starts counting its own steps.
- The arming at `reset` stays as the provisional bound, so a frame zero that never lands is still caught — without it, an observation channel that never delivers would remove the trial's deadline entirely.

This affects every env-server sim (libero, robolab) equally; their timeout margins were absorbing the offset silently.

## Test plan

`test_trial_budget_starts_at_the_first_complete_observation` — frame zero at 0.02 and a marked observation at 0.06 against a 0.05 budget. Anchored at `reset` the trial expires at 0.05 and the policy never sees the mark; anchored at frame zero it runs to 0.07 and does. Verified it fails on `main` and passes here.

Full suite: 1016 passed, 8 skipped. One pre-existing failure unrelated to this change — `utilities/tests/test_check_basedpyright_baseline_ratchet.py::test_build_rename_map_includes_staged_rename` raises `CalledProcessError` from git, and reproduces with these changes stashed (it appears to dislike running in a linked worktree).

### CODE REVIEW

File: positronic/policy/harness.py [hunk 1]
```diff
@@ -442,9 +443,13 @@ def _step(self, clock: pimm.Clock) -> Generator[pimm.Sleep, None, None]:
         The session output already carries absolute timestamps (stamped by the
         outermost scheduling wrapper). The harness only demuxes by channel.
         """
+        awaiting_frame_zero = bool(self._awaiting_obs)  # ``_build_obs`` clears it, so read it first
```
[the remark this answers is not in the document]
Comment (vertix) [reply positronic/policy/harness.py]: Fixed in 60f25024 — you're right, and it hits the common path rather than an edge case.

`_awaiting_obs` tracks *delivery*, not *usability*: `_build_obs` discards a channel the moment a message arrives (line 402), then bails out at the serializer check (line 407) with the discard already applied. So a first delivery whose serializer rejects the sample — `robot_state` while the arm is `RESETTING`, which is the state a real embodiment is in right after a reset — leaves the set empty without ever yielding an observation, and the anchor never fires.

Replaced the inference with an explicit `_budget_anchored` flag, cleared in `_begin_episode` and set on the first `_build_obs` that actually returns an observation. Same "only the first lands" shape as `TelemetryScope.start_rollout`, which anchors span durations at the same instant.

`test_trial_budget_starts_at_the_first_usable_observation` now pins it: the delivery at 0.02 is rejected by the camera serializer, so the first usable observation is at 0.04. I verified the test fails both against `main` and against the previous `_awaiting_obs` version, and passes with the flag.

File: positronic/policy/harness.py [hunk 2]
```diff
@@ -442,9 +443,13 @@ def _step(self, clock: pimm.Clock) -> Generator[pimm.Sleep, None, None]:
...
         obs = self._build_obs(clock)
         if obs is None:
             return
+        if awaiting_frame_zero and self._task is not None:
+            # An env that paces the trial counts its own steps from frame zero, so the budget must too.
+            self._deadline = clock.now() + self._task.timeout
```
[the remark this answers is not in the document]
Comment (vertix) [reply positronic/policy/harness.py]: The mechanism you describe is real, but the behaviour is intended, so I'm leaving it — flagging the reasoning for a human to overrule if they disagree.

The deadline armed at `reset` is a safety net, not a first attempt at the budget. An earlier revision of this change left it unarmed until the first observation, exactly so no trial could expire before its episode began. That version hangs: an observation channel that never delivers removes the trial's deadline altogether, and a run with a misconfigured embodiment never terminates. The provisional arming is what closes that hole, so it has to be enforceable before the first observation arrives.

Given that, a trial whose startup consumes its entire `timeout` expiring without inference is the safety net doing its job — if an env needs longer than the whole trial budget to produce one observation, the budget is misconfigured or the env is broken, and failing loudly beats silently extending. The window you identify is only the boundary round where the first observation and the expiry land together; below that, the re-anchor fires normally and the trial gets its full budget.

Closing that last round would mean building observations before the terminal check in `_run`, which breaks the documented one-action-per-round dispatch (and `_build_obs` has side effects, so it can't simply be called twice). That cost isn't worth a boundary case that only arises when startup already ≈ the entire budget.

Leaving this thread open for a human to weigh in.

File: positronic/policy/tests/test_harness.py [hunk 3]
```diff
@@ -610,6 +610,43 @@ def test_trial_timeout_self_terminates(world):
...
+    as a marker. Measured from the reset the budget ends at 0.05 and the policy never sees the marker;
+    measured from the first usable observation it ends at 0.09 and the policy does.
+
+    The rejected delivery matters on its own: it clears the channels out of ``_awaiting_obs`` without
+    yielding an observation, so a budget anchored on that set alone would never move off the reset.
+    """
+    policy = SpyPolicy()
+    embodiment = make_embodiment()
+    usable = iter([None])  # the first camera sample is not ready; every later one is
+    embodiment.observations['image.cam'] = Observation(
```
[the remark this answers is not in the document]
Comment (vertix) [reply positronic/policy/tests/test_harness.py]: Fixed in 16bc84aa.

Hoisted `CAM = 'image.cam'` to module scope in `test_harness.py` and migrated all 14 sites — `make_embodiment`'s default, `_pair_all`, and every test pairing against the channel. `test_sampled_e2e.py` was the strongest case for it: it already imports `make_embodiment` from here and then re-spelled the literal itself, so the contract crossed files invisibly. It now imports `CAM`.

I migrated the whole file rather than only my own line: a constant used at one site while thirteen literals remained would have been worse than either extreme.

File: positronic/policy/tests/test_harness.py [hunk 4]
```diff
@@ -35,6 +35,10 @@
 from positronic.policy.wrappers import ChunkedSchedule
 from positronic.tests.testing_coutils import ManualDriver, RecordingEmitter, drive_scheduler
 
+# The camera channel every embodiment these tests build carries, named by the producer and the tests that
+# pair against it alike.
```
[the remark this answers is not in the document]
Comment (vertix) [reply positronic/policy/tests/test_harness.py]: Fixed in db850c52 — agreed, the comment listed who happens to use the constant, which is the thing `diff-comments` rules out. `CAM = 'image.cam'` says it on its own. Removed.

File: positronic/policy/tests/test_harness.py [hunk 5]
```diff
@@ -35,6 +35,10 @@
...
+CAM = 'image.cam'
```
[the remark this answers is not in the document]
Comment (vertix) [reply positronic/policy/tests/test_harness.py]: Fixed in db850c52. `CAM` now sits directly above `make_embodiment`, the first thing that establishes the channel, instead of ahead of the unrelated `_eval_pass` helper.
